tests/process.rs

```diff
@@ -378,3 +378,99 @@ fn test_refresh_process() {
     // Checks that the process is still listed.
     assert!(s.process(pid).is_some());
 }
+
+#[test]
+fn test_wait_child() {
+    if !sysinfo::System::IS_SUPPORTED || cfg!(feature = "apple-sandbox") {
+        return;
+    }
+    let p = if cfg!(target_os = "windows") {
+        std::process::Command::new("waitfor")
+            .arg("/t")
+            .arg("300")
+            .arg("RefreshProcess")
+            .stdout(std::process::Stdio::null())
+            .spawn()
+            .unwrap()
+    } else {
+        std::process::Command::new("sleep")
+            .arg("300")
+            .stdout(std::process::Stdio::null())
+            .spawn()
+            .unwrap()
+    };
+
+    let pid = Pid::from_u32(p.id() as u32);
+
+    let mut s = sysinfo::System::new();
+    s.refresh_process(pid);
+    let process = s.process(pid).expect("Process not found!");
+
+    // Kill the child process.
+    process.kill();
+    // Wait for child process should work.
+    let wait_pid = process.wait().unwrap();
+    assert_eq!(wait_pid, pid);
+    // Let's give some time to the system to clean up...
+    std::thread::sleep(std::time::Duration::from_secs(1));
+
+    // Child process should not be present.
+    assert!(!s.refresh_process(pid));
+}
+
+#[test]
+#[ignore]
```

Comment:
Any suggestions for testing non child wait process?

For linux:
I tried the `(...)` approach that starts nested process and ends the middle process creating a non child process.
Then I search the process by name, it works okay if I run the test alone, but fails if we run a lot of test parallely because there are a lot of `sleep` processes running which makes sense.

I tried creating the same but with a custom name using `exec -a custom_sleep sleep 2 &` which doesn't work as expected.

For windows:
Not sure :D

Reply:
The simplest solution is to get the PID from the `Command` API and then retrieve the process with it like that there is no conflict. Also, to test "non-child", on linux at least there is `setsid` iirc. On windows since there is no difference, there is no point in testing it.